test/browsers/wdio-mock.ts

```diff
@@ -0,0 +1,27 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import type { RemoteOptions } from 'webdriverio';
+
+class FakeWebdriver {
+  isMobile = true;
+  options: RemoteOptions;
+
+  constructor(options: RemoteOptions) {
+    this.options = options;
+  }
+
+  setTimeout() {
+    // noop
+  }
+}
+
+export const remoteMock = jest.fn();
+
+beforeEach(() => {
+  remoteMock.mockReset();
+  remoteMock.mockImplementation((options: RemoteOptions) => new FakeWebdriver(options));
+});
+
+jest.mock('webdriverio', () => ({
+  remote: remoteMock,
+}));
```

Comment:
Adding this mock is probably the biggest change in this PR

Previously  tests called private methods directly and asserted return value.

The new strategy respects typescript and calls only public API. Then we use this module mock to assert that webdriver received expected parameters and capabilities